scripts/sct_deepseg.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+"""
+This command-line tool is the interface for the deepseg API that performs segmentation using deep learning from the
+ivadomed package.
+"""
+
+# TODO: Add link to example image so users can decide wether their images look "close enought" to some of the proposed
+#  models (e.g., mice, etc.).
+# TODO: add test to make sure that all postprocessing flags match the core.DEFAULT dictionary items
+# TODO: Fetch default value (and display) depending on the model that is used.
+
+from __future__ import absolute_import
+
+import sys
+import os
+import argparse
+import colored
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.deepseg.core
+import spinalcordtoolbox.deepseg.models
+import spinalcordtoolbox.utils
+
+from sct_utils import init_sct, printv, display_viewer_syntax
+
+
+def get_parser():
+
+    parser = argparse.ArgumentParser(
+        description="Segmentation using deep learning.",
+        add_help=None,
+        formatter_class=sct.utils.SmartFormatter,
+        prog=os.path.basename(__file__).strip(".py"))
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        "-i",
+        help="Image to segment.",
+        metavar=sct.utils.Metavar.file)
```

Comment:
I know this is what lots of apps do (`ffmpeg` sticks out to me) but I think having mandatory-options is a oxymoron.

> There is generally some confusion among authors of command line scripts of when to use which, so here is a quick overview of the differences. As its name indicates, an option is optional. While arguments can be optional within reason, they are much more restricted in how optional they can be.

- https://click.palletsprojects.com/en/7.x/parameters/

Also, it's not truly mandatory: if you use `-list-models` or `-install-model` then `-i` is unused.

Reply:
> In my ideal world the signature would be sct_deepseg {[-keep-largest-object] model input_image [output_image],-install-model model,-list-models}. Another common convention (and I think what the other SCT scripts use?) is [-o output_image].

i know, there are some inconsistencies but i feel this is out of the scope of this PR and would require consensus. Let me look at your other comments and i might come back to this.

> I think specifying an output suffix -- specifying only part of the filename -- is difficult to work with in a unix world even though it seems like the minimal amount of information to give. Just use full filenames always and the tool will feel at home on the command line. 

yup, this is now implemented

> And consider using subfolders: instead of adding a suffix (in the middle of the filename), use seg/image.nii.gz or seg/category1/image.nii.gz.

this was broken if the directory does not exist. I am now created the folder: 10628f85203090278ed782bf21dcef063c1482ae